File: DSA_Codesheet/C/remove_node.c
#include <stdio.h>
#include <stdlib.h>

struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode *removeNthFromEnd(struct ListNode *head, int n) {
    struct ListNode *dummy = (struct ListNode *)malloc(sizeof(struct ListNode));
    dummy->next = head;
    struct ListNode *first = dummy;
    struct ListNode *second = dummy;

    
    for (int i = 0; i <= n; i++) {
        first = first->next;
    }

    
    while (first != NULL) {
        first = first->next;
        second = second->next;
    }

 
    struct ListNode *temp = second->next;
    second->next = temp->next;
    free(temp);

    return dummy->next; 
}

int main() {
    
    struct ListNode *head = (struct ListNode *)malloc(sizeof(struct ListNode));
    head->val = 1;
    head->next = NULL;

   
    int n = 1; // Change this to the desired n value
    struct ListNode *newHead = removeNthFromEnd(head, n);

  
    struct ListNode *current = newHead;
    while (current != NULL) {
        printf("%d ", current->val);
        current = current->next;
    }

    return 0;
}

File: DSA_Codesheet/Python/remove_node.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def removeNthFromEnd(head, n):
    dummy = ListNode(0)
    dummy.next = head
    first = dummy
    second = dummy

    
    for _ in range(n+1):
        first = first.next

    
    while first:
        first = first.next
        second = second.next

    second.next = second.next.next

    return dummy.next 
